### PR TITLE
Preparing for a release v0.3

### PR DESCRIPTION
**Pull request by chaithyagr**
This PR holds the release for v0.3.
We will have pyNFFT as an optional install clearing the way for easier install for users. The README is appropriately updated.
Also, I have added the current maintainers/developers in the README.

The updates in this release include:

1) #118 Refactor and remove unwanted imports in CircleCI
2) #111 Fix FFT class, move to `scipy.fft` to better handle multi-process. Also fix the grid-search
3) #113 Fix the `normalize_frequency_locations` code to work well with given `KMax`
4) #114 Add oscar regularization based on `scale`

### CODE REVIEW

**Comment by chaithyagr** [comment]
@sfarrens let me know if you have issues in running things on Mac. I will try to see if I can get a CI up for mac (currently I didnt considering that our tests are extensive and take a lot of time)

File: README.rst [hunk 1]
```diff
@@ -18,21 +18,46 @@ pySAP-mri
...
+``pip install pynfft2``
+
+Mac:
+````
+
+``brew tap brewsci/science``
+
+``brew install nfft``
+
+``pip install pynfft``
```
Comment (sfarrens) [inline README.rst:59]: @chaithyagr I couldn't get this to work. But there is a functioning conda package.

```
conda install -c conda-forge pynfft
```

I would suggest we put this for now (provided there is nothing special in pyNFFT2 that is required) and we can open an issue to find something better for macOS for the next release.

**Comment by sfarrens** [comment]
Hi @chaithyagr this branch builds fine on my mac so I think everything should be fine to release. 

I made one comment regarding the installation of pyNFFT on macOS. If pyNFFT will work then I suggest you make the proposed change. If not, then I would recommend we simply remove the macOS section until we have a stable solution. Hopefully by the next release.

File: README.rst (hunk 1 above)
Comment (chaithyagr) [reply README.rst:59]: Oh no no, `pynfft` does not build well in linux. Also, last I checked the earlier pynfft had some SEGFAULT issues (i dont remember exactly)

**Review by sfarrens — APPROVED** [review]

**Review by zaccharieramzi — APPROVED** [review]
LGTM

**Comment by chaithyagr** [comment]
Currently, the tests are failing due to issues in Modopt. @sfarrens can you please make a release of Modopt so that I can make a release here?

**Comment by sfarrens** [comment]
@chaithyagr I am waiting for a review of the ModOpt release PR then I will release a new patch.

**Comment by sfarrens** [comment]
@chaithyagr ModOpt v1.5.1 has been released.

**Pull request merged**